CRM/Odoosync/Sync/Contribution/PendingContribution.php

```diff
@@ -41,26 +52,26 @@ public function __construct() {
    * @return array
    */
   public function getIds() {
-    try {
-      $contributionList = civicrm_api3('Contribution', 'get', [
-        'return' => ["id"],
-        'is_deleted' => ['IS NOT NULL' => 1],
-        'options' => ['limit' => (int) CRM_Odoosync_Sync_BatchSize::getCurrentBatchSize()],
-        'custom_' . $this->syncStatusFieldId => $this->syncStatusValue,
-      ]);
-
-      $contributionListId = [];
-      foreach ($contributionList['values'] as $contribution) {
-        $contributionListId[] = $contribution['contribution_id'];
-      }
+    $query = "
+      SELECT 
+        contribution.id AS id 
+      FROM civicrm_contribution AS contribution
+      LEFT JOIN odoo_invoice_sync_information AS info 
+        ON contribution.id = info.entity_id
+      WHERE (info.do_not_sync IS NULL OR info.do_not_sync != 1)
```

Comment:
hmm I though there is a way in civicrm API to avoid this but I did some checking and it appear that there is no way. 

So sorry to say this but can u please revert the last commit, since you are setting the value for this field to 0 (No) by default I think it is fine to keep it required and keep using the API instead of the direct SQL query.